src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs

```diff
@@ -372,7 +372,73 @@ public override StackValue Execute(ProtoCore.Runtime.Context c, List<StackValue>
+                        StackValue svFalse = formalParameters[2];
 
+                        // If run in delta execution environment, we don't 
+                        // create language blocks for true and false branch, 
+                        // so directly return the value.
+                        if (runtimeCore.Options.GenerateSSA)
+                            return svCondition.BooleanValue ? svTrue : svFalse;
+
+                        Validity.Assert(svTrue.IsInteger);
+                        Validity.Assert(svFalse.IsInteger);
+                        int blockId = svCondition.BooleanValue ? (int)svTrue.IntegerValue : (int)svFalse.IntegerValue;
+                        int oldRunningBlockId = runtimeCore.RunningBlock;
+                        runtimeCore.RunningBlock = blockId;
+
+                        int returnAddr = stackFrame.ReturnPC;
+
+                        int ci = ProtoCore.DSASM.Constants.kInvalidIndex;
+                        int fi = ProtoCore.DSASM.Constants.kInvalidIndex;
+                        if (interpreter.runtime.rmem.Stack.Count >= ProtoCore.DSASM.StackFrame.StackFrameSize)
+                        {
+                            ci = stackFrame.ClassScope;
+                            fi = stackFrame.FunctionScope;
+                        }
+
+                        // The class scope does not change for inline conditional calls
+                        StackValue svThisPtr = stackFrame.ThisPtr;
+
+
+                        int blockDecl = 0;
+                        int blockCaller = oldRunningBlockId;
+                        StackFrameType type = StackFrameType.LanguageBlock;
+                        int depth = (int)interpreter.runtime.rmem.GetAtRelative(StackFrame.FrameIndexStackFrameDepth).IntegerValue;
+                        int framePointer = rmem.FramePointer;
+
+                        // Comment Jun: Calling convention data is stored on the TX register
+                        StackValue svCallconvention = StackValue.BuildCallingConversion((int)ProtoCore.DSASM.CallingConvention.BounceType.Implicit);
+                        interpreter.runtime.TX = svCallconvention;
+
+                        List<StackValue> registers = interpreter.runtime.GetRegisters();
+
+                        // Comment Jun: the caller type is the current type in the stackframe
+                        StackFrameType callerType = stackFrame.StackFrameType;
+
+
+                        blockCaller = runtimeCore.DebugProps.CurrentBlockId;
+                        StackFrame bounceStackFrame = new StackFrame(svThisPtr, ci, fi, returnAddr, blockDecl, blockCaller, callerType, type, depth, framePointer, 0, registers, 0);
+
+                        ret = interpreter.runtime.Bounce(blockId, 0, bounceStackFrame, 0, false, runtimeCore.CurrentExecutive.CurrentDSASMExec, runtimeCore.Breakpoints);
+
+                        runtimeCore.RunningBlock = oldRunningBlockId;
+                        break;
+                    }
```

Comment:
This entire logic seems to be identical to the one for the inline conditional builtin. If so, then you can refactor the code into a helper function that can be called from both places to eliminate duplicate code.